hpedockerplugin/request_context.py

```diff
@@ -0,0 +1,567 @@
+import abc
+import json
+import re
+import six
+from collections import OrderedDict
+
+from oslo_log import log as logging
+
+import hpedockerplugin.exception as exception
+from hpedockerplugin.hpe import volume
+from hpedockerplugin.hpe import share
+
+LOG = logging.getLogger(__name__)
+
+
+class RequestContextBuilderFactory(object):
+    def __init__(self, all_configs):
+        self._all_configs = all_configs
+
+        # if 'block' in all_configs:
+        #     block_configs = all_configs['block']
+        #     backend_configs = block_configs[1]
+        #     self._vol_req_ctxt_creator = VolumeRequestContextBuilder(
+        #         backend_configs)
+        # else:
+        #     self._vol_req_ctxt_creator = NullRequestContextBuilder(
+        #         "ERROR: Volume driver not enabled. Please provide hpe.conf "
+        #         "file to enable it")
+
+        if 'file' in all_configs:
+            file_configs = all_configs['file']
+            f_backend_configs = file_configs[1]
+            self._file_req_ctxt_builder = FileRequestContextBuilder(
+                f_backend_configs)
+        else:
+            self._file_req_ctxt_builder = NullRequestContextBuilder(
```

Comment:
is `NullRequestContextBuilder` still required, since we are going to have one unified `hpe.conf` instead of `hpe.conf` and `hpe_file.conf` ?

Reply:
Yes. It will be required for cases where user configures ONLY volume or ONLY File Persona.